.github/workflows/function_app_deploy.yaml

```
@@ -123,15 +129,25 @@ jobs:
           tenant-id: ${{ env.ARM_TENANT_ID }}
           subscription-id: ${{ env.ARM_SUBSCRIPTION_ID }}
 
+      - name: Deploy
+        uses: azure/webapps-deploy@v2
+        if: ${{ env.USE_STAGING_SLOT == 'false' }}
```

isn't it

```suggestion
        if: ${{ env.USE_STAGING_SLOT == false }}
```

without quotes?